compiler/hash-ast/src/origin.rs

```diff
@@ -44,3 +44,44 @@ impl Display for BlockOrigin {
         }
     }
 }
+
+/// Denotes where a pattern was used as in the parent of the pattern. This is
+/// useful for propagating errors upwards by signalling what is the current
+/// parent of the pattern. This only contains patterns that can be compound
+/// (hold multiple children patterns).
+#[derive(Clone, Copy, Debug)]
+pub enum PatOrigin {
+    /// The parent is a tuple, i.e `(x, 1, 2)`
+    Tuple,
+    /// The parent is a pseudo-pattern originating as a named field of some
+    /// other parent pattern. Named fields could occur within tuples or
+    /// constructor patterns, i.e `Bar(max = 123)`, where the `max` would be
+    /// the named field.
+    NamedField,
```

Comment:
What if the parent is a positional field? How is this different from `Tuple`/`Constructor`?

Reply:
Could you add this as a comment?